shared/reducers/unlock-folders.js

```diff
@@ -8,29 +8,54 @@ import type {UnlockFolderActions, Device} from '../constants/unlock-folders'
 export type State = {
   phase: 'dead' | 'promptOtherDevice' | 'paperKeyInput' | 'success',
   devices: ?Array<Device>,
-  paperkey: ?HiddenString,
   paperkeyError: ?HiddenString
 }
 
 const initialState: State = {
   phase: 'dead',
   devices: null,
-  paperkey: null,
   paperkeyError: null
 }
 
 export default function (state: State = initialState, action: UnlockFolderActions): State {
   // TODO: Fill out the rest of this reducer
   switch (action.type) {
-    case Constants.toPaperKeyInput:
+    case Constants.loadDevices:
       if (action.error) {
         return state
       } else {
+        const devices = action.payload.devices.map(({name, type, deviceID}) => ({
+          type: type === 'mobile' ? 'mobile' : 'desktop',
```

Comment:
The type in the avdl for device.type is string. I forgot that paperkey can show up here. I'll change this and add a proper type for DeviceType.

the reason I had to do this is because I need to convert from string type to an enum type.